model/llm/llm.go

```diff
@@ -32,7 +32,7 @@ type Model struct {
 }
 
 // NewModel returns an LLM model corresponding to the given identifier which is queried via the given provider.
-func NewModel(provider provider.Query, modelIdentifier string) model.Model {
+func NewModel(provider provider.Query, modelIdentifier string) *Model {
```

Comment:
Not a huge thing but why is this necessary?

Reply:
Because in the tests we create a model with `NewModel` and then call `GenerateTestsFor....` and this is not a method of `model.Model` anymore.